### PR TITLE
fix(#34): don't spawn more than one git process per file

### LINKED ISSUE #34: several 'git blame' spawned when scroll the cursor at plugin load
If I toggle git-blame and immediately move up and down the line with the cursor, I am experiencing several ```git blame``` command spawned by the plugin.
The commands do not ever resolve and the CPU gets used at 100%.
The only solution is to quit nvim.

Any idea why this is happening? I may cook a patch if pointed in the right direction.

### PR DESCRIPTION
**Pull request by grnnja**
Fixes #34 

In load_blames, check if we are already loading from the file and don't try
loading again if we are.

This fixes git-blame.nvim spawning a bunch of git processes if the first
one is slow

### CODE REVIEW

**Comment by f-person** [comment]
Awesome stuff, thank you!

File: lua/gitblame/init.lua [hunk 1]
```diff
@@ -115,6 +122,9 @@ local function load_blames(callback)
             on_stdout = function(data)
                 process_blame_output(blames, filepath, data)
                 if callback then callback() end
+            end,
+            on_exit = function()
+                files_data_loading[filepath] = false
```
Comment (f-person) [inline lua/gitblame/init.lua:127]: Please set it to `nil` instead in order to remove it from the table:

```lua
files_data_loading[filepath] = nil
```

**Review by f-person — CHANGES_REQUESTED** [review]

File: lua/gitblame/init.lua [hunk 2]
```diff
@@ -12,6 +12,9 @@ local last_position = {}
 ---@type table<string, table>
 local files_data = {}
 
+---@type table<string, table>
```
[the remark this answers is not in the document]
Comment (grnnja) [reply lua/gitblame/init.lua]: Fixed!

File: lua/gitblame/init.lua (hunk 1 above)
Comment (grnnja) [reply lua/gitblame/init.lua]: Fixed!

**Review by f-person — APPROVED** [review]

**Pull request merged**